Sources/NIO/CircularBuffer.swift

```diff
@@ -692,3 +692,17 @@ extension CircularBuffer {
         return self.verifyInvariants() && self.unreachableAreNil()
     }
 }
+
+extension CircularBuffer: Equatable where Element: Equatable {
+    public static func ==(lhs: CircularBuffer, rhs: CircularBuffer) -> Bool {
+        lhs.count == rhs.count && zip(lhs, rhs).allSatisfy(==)
```

Comment:
We need Swift 5 compatibility, so sadly for you the `return` statement cannot be elided here.

Reply:
Fixed! 🙂